warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__trip_id_alignment.sql

```diff
@@ -0,0 +1,56 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__rt_feed_guideline_index') }}
+),
+
+fct_daily_rt_feed_files AS (
+    SELECT * FROM {{ ref('fct_daily_rt_feed_files') }}
+),
+
+fct_daily_rt_feed_validation_notices AS (
+    SELECT * FROM {{ ref('fct_daily_rt_feed_validation_notices') }}
+),
+
+count_files AS (
+    SELECT
+        base64_url,
+        date,
+        SUM(parse_success_file_count) AS rt_files
+    FROM fct_daily_rt_feed_files
+    GROUP BY 1, 2
+),
+
+critical_notices AS (
+    SELECT
+        date,
+        base64_url,
+        SUM(total_notices) AS errors,
+    FROM fct_daily_rt_feed_validation_notices
+    -- Description for code E003:
+    ---- "All trip_ids provided in the GTFS-rt feed must exist in the GTFS data, unless the schedule_relationship is ADDED"
+    WHERE code = "E003"
+    GROUP BY 1, 2
+),
+
+int_gtfs_quality__trip_id_alignment AS (
+    SELECT
+        idx.date,
+        idx.base64_url,
+        idx.feed_type,
+        {{ trip_id_alignment() }} AS check,
+        {{ fixed_route_completeness() }} AS feature,
+        rt_files,
+        errors,
+        CASE
+            WHEN COALESCE(errors, 0) = 0 THEN "PASS"
+            WHEN rt_files IS NOT NULL THEN "FAIL" -- we had no errors, but did have files
```

Comment:
A bit confused by the comment here.... If `COALESCE(errors,0) != 0` then there *were* errors, right?

I also think this may need adjustment -- should the `rt_files` check be part of the first condition? I think as written that `rt_files IS NULL` will lead to `PASS` under this logic (because presumably if `rt_files` is null then `errors` will be null and then will be `COALESCED` to 0, if I am understanding correctly.)

Not to add more work to your plate, but I am wondering if at some point it will be necessary to audit null handling in the different checks and standardize at all 😅

Reply:
No surprise that you are right. I borrowed this from the rt check that Andrew wrote...but what I really should have done is caught the issue in his PR! I've now updated both - erring on the side of specificity so that it's clear what the conditions are to pass and fail.

To pass: at least one RT file exists, and no errors detected
To fail: at least one RT file exists, more than zero errors detected
To have a null value: no RT files exist. Note that there are none of these results now, since the index table must not have a row for days when no RT files exist.

Reply:
Yeah, I reviewed his PR too 🙈 Thanks for updating both!